dnf/comps.py

```diff
@@ -507,78 +507,74 @@ def filter(pkgs):
         return pkgs
 
     def _removable_pkg(self, pkg_name):
-        prst = self.persistor
-        count = 0
-        if self._reason_fn(pkg_name) != 'group':
-            return False
-        for id_ in prst.groups:
-            p_grp = prst.group(id_)
-            count += sum(1 for pkg in p_grp.full_list if pkg == pkg_name)
-        return count < 2
+        return self.persistor.removable_pkg(pkg_name)
 
     def _removable_grp(self, grp_name):
         prst = self.persistor
         count = 0
-        if not prst.group(grp_name).installed:
+        if not prst.group(grp_name).is_installed:
             return False
-        for id_ in prst.environments:
-            p_env = prst.environment(id_)
-            count += sum(1 for grp in p_env.full_list if grp == grp_name)
+        for id_ in prst.environments():
```

Comment:
`prst.environments()` actually does that... I don't know why it's like this... Fixed